core/src/main/java/org/apache/calcite/rel/rules/SubQueryRemoveRule.java

```diff
@@ -910,6 +910,24 @@ private static void matchJoin(SubQueryRemoveRule rule, RelOptRuleCall call) {
     int nFieldsLeft = join.getLeft().getRowType().getFieldCount();
     int nFieldsRight = join.getRight().getRowType().getFieldCount();
 
+    // Correlation columns are also should be considered.
+    // For example:
+    //                                   LogicalJoin
+    //              left                                          right
+    //                |                                             |
+    // LogicalProject,inputs=0..1)                           LogicalValues.NONE.[0]
```

Comment:
`LogicalProject,inputs=0..1)` <-- seems incorrectly formatted, can you check?

Reply:
done